Download the dataset with 10000 taxi trips below: sample10000.csv

## Using the data dictionary, check how many passengers in the sample paid for their ride with cash.

In [1]:
import pandas as pd
import numpy as np

import scipy as sp
import scipy.stats


%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
sample10000 = pd.read_csv('sample10000.csv', parse_dates=[1,2])

In [3]:
sample10000.head()

VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0         2  2016-12-19 20:13:04   2016-12-19 20:16:58                1   
1         1  2016-12-27 19:28:48   2016-12-27 19:30:31                1   
2         1  2016-12-17 00:57:37   2016-12-17 01:10:37                3   
3         1  2016-12-01 20:32:54   2016-12-01 20:52:17                1   
4         2  2016-12-12 19:25:14   2016-12-12 19:34:15                2   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0           0.87           1                  N           238           142   
1           0.10           1                  N           231           144   
2           2.30           1                  N            90           233   
3           5.00           1                  N            13           143   
4           1.43           1                  N           164            79   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0             1          5.0    0.5      0.5        1.50           0.0   
1             2          3.0    1.0      0.5        0.00           0.0   
2             1         11.0    0.5      0.5        2.45           0.0   
3             1         18.5    0.5      0.5        3.95           0.0   
4             1          7.5    1.0      0.5        1.00           0.0   

   improvement_surcharge  total_amount  
0                    0.3          7.80  
1                    0.3          4.80  
2                    0.3         14.75  
3                    0.3         23.75  
4                    0.3         10.30

In [4]:
sample10000.payment_type.value_counts()

1    6397
2    3543
3      48
4      12
Name: payment_type, dtype: int64

In [5]:
sample10000[sample10000.payment_type == 2].shape[0]

3543

**Correct** 3543

## Build a 99% confidence interval for the proportion of cash payers. What is its' lower boundary?

Estimating the proportion of cachers

In [6]:
is_cache = sample10000.payment_type == 2
is_cache.mean()

0.3543

Standard deviation

In [7]:
from statsmodels.stats.proportion import proportion_confint
res = proportion_confint(sum(is_cache), len(is_cache), alpha=0.01)

print (res)
res[0]

(0.34197979174644566, 0.36662020825355435)


0.34197979174644566

**Correct** 0.34197979174644566

## Use the same sample to estimate the average trip distance in miles. Provide the answer with at least two digits after decimal.

In [13]:
print sample10000.trip_distance.mean()
print round(sample10000.trip_distance.mean(), 2)
print round(sample10000.trip_distance.median(), 2)


2.881264
2.88
1.65


**Correct** 2.881264

## What is the standard deviation of the estimator from the previous question? Provide the answer with at least three digits after decimal.

In [9]:
s = sample10000.trip_distance.std(ddof=1) / np.sqrt(len(sample10000.trip_distance))

round(s, 3)

0.036

**Correct** 0.036

## Calculate 95% confidence interval for the mean trip distance. What is the upper boundary? Provide the answer with at least two digits after decimal.

In [10]:
from statsmodels.stats.weightstats import _tconfint_generic
res = _tconfint_generic(
    sample10000.trip_distance.mean(), s, len(sample10000.trip_distance) - 1, 0.05, 'two-sided')

print (res)
round(res[1], 2)



(2.8110118487985924, 2.9515161512013965)


2.95

**Correct** 2.95